README.md

```diff
@@ -31,7 +31,14 @@ Using AWS SNS or Google PubSub it is possible to have streaming notifications wh
 
 For existing data in buckets, the SNS or PubSub notifications may be simulated such that the indexing functions are started for each object in the bucket. This is useful because only a single code path is necessary for indexing the contents of an object.
 
-3. Using the Indexd REST API for record insertion.
+3. Indexing void object for fully control the bucket structure.
+
+Indexd supports void or blank records that allows users to pre-register data files in indexd before actually registering them. The complete flow contains three main steps: pre-register, hash/size/url populating and data node registration:
+    - Fence requests blank object from indexd. Indexd creates an object with no hash, size, and urls except the `uploader` field.
+    - Indexd listener mornitors bucket update, update to indexd with url, hash, size.
+    - Windmill lists indexd records that’s owned by the user that has empty `acl`. The user fills all empty fields and submit the request to indexd to update the `acl`.
+
```

Comment:
could you add a link to the data upload docs? https://github.com/uc-cdis/cdis-wiki/tree/master/dev/gen3/data_upload